features/support/utils_gen.rb

```diff
@@ -0,0 +1,222 @@
+require 'net/http'
+require 'uri'
+
+
+def set_current_user(user = nil)
+  if user.nil?
+    begin
+      case ENV["platform"]
+      when "android"
+        $current_user = Members.new.get("android")
+      when "ios", "ios-simulator"
+        $current_user = Members.new.get("ios")
+      else
+        puts "- current_user is not set. Platform should be one of:\n * android\n * ios\n * ios-simulator"
+      end
+    end
+  else
+    $current_user = Members.new.get(user)
+  end
+end
+
+def embed_image(name, take_screenshot = true)
+  screenshot_path = File.dirname(__FILE__) + '/media/' + name + '.png'
+
+  if take_screenshot
+    screenshot(screenshot_path)
+  end
+  embed(screenshot_path, "image/png", "SCREENSHOT")
+end
+
+
+def embed_screenshot scenario
+  name = scenario.name.gsub(" ", "_").downcase
+  screenshot_path = File.dirname(__FILE__) + '/media/' + ENV['platform'] + '/'
+  screenshot_embed(options = {:prefix => screenshot_path, :name => name})
+end
+
+def wait_until_loader_disappear
+  while query("* id:'wait_spinner'").size != 0
+    sleep 0.5
+  end
+end
+
+def wait_until_text_disappear(text)
+  while check_if_text_exists(text)
+    sleep 0.5
+  end
+end
+
+def wait_until_text_visible(text)
+  wait_until_element_ready "* {text CONTAINS[C] '#{text}'}"
+end
+
+def wait_until_element_ready(qr)
+  10.times do
+    if query(qr).size > 0
+      break
+    else
+      sleep 1
+      puts "- Element is not ready yet ...."
+    end
+  end
+end
+
+def wait_until_one_element_ready(*args)
+  # this will wait until one of the element visible in the given query list
+  # wait_until_one_element_ready("* text:'Stores'", "* text:'Places'")
+
+  element = :not_exist
+
+  10.times do
+    args.each do |qr|
+      if query(qr).size > 0
+        element = :exist
+        break
+      end
+    end
+
+    if element == :exist
```

Comment:
you don't need this check as well as `element = :exist` assignment, if this didn't break before then the element doesn't exist anyways

Reply:
Updated this function.